sdk/go/common/workspace/creds.go

```diff
@@ -166,6 +166,23 @@ func GetStoredCredentials() (Credentials, error) {
 		return Credentials{}, err
 	}
 
+	// If the file is empty, we can act as if it doesn't exist rather than trying
+	// (and failing) to deserialize its contents. This allows us to recover from
+	// situations where a write to the file was interrupted or it was otherwise
+	// clobbered.
+	stat, err := os.Stat(credsFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return Credentials{}, nil
+		}
+		return Credentials{}, fmt.Errorf("reading '%s': %w", credsFile, err)
+	}
+
+	if stat.Size() == 0 {
+		return Credentials{}, nil
+	}
+
+	// The file isn't empty -- we'll try to read and deserialize its contents.
```

Comment:
Why stat instead of just seeing if `len(c) == 0` below?

Reply:
God I'm an idiot. Much better; thanks :facepalm: